rules/ast/functions.go

```diff
@@ -444,10 +494,11 @@ var functions = map[string]*Function{
 		callFn:     countScalarImpl,
 	},
 	"delta": {
-		name:       "delta",
-		argTypes:   []ExprType{MatrixType, ScalarType},
-		returnType: VectorType,
-		callFn:     deltaImpl,
+		name:         "delta",
+		argTypes:     []ExprType{MatrixType},
```

Comment:
We should still check the type of the optional argument if it gets passed in. Otherwise you'll get panics like:

    interface conversion: *ast.StringLiteral is not ast.ScalarNode: missing method Eval